libsmtutil/CHCSmtLib2Interface.cpp

```diff
@@ -64,50 +63,55 @@ void CHCSmtLib2Interface::registerRelation(Expression const& _expr)
 	if (!m_variables.count(_expr.name))
 	{
 		auto fSort = dynamic_pointer_cast<FunctionSort>(_expr.sort);
-		string domain = m_smtlib2->toSmtLibSort(fSort->domain);
+		string domain = toSmtLibSort(fSort->domain);
 		// Relations are predicates which have implicit codomain Bool.
 		m_variables.insert(_expr.name);
 		write(
-			"(declare-rel |" +
+			"(declare-fun |" +
```

Comment:
Why `(declare-fun |expr| ...)` and not `(declare-fun expr ...)`?

Reply:
Because afaik `|...|` works as quotes in smtlib2 and it accepts more symbols inside `|...|`.